<html>
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <meta name="description" content="A collection of notes describing a few aspects of R for bioinformatics.">
  <meta name="viewport" content="width=device-width,initial-scale=1">

  <link rel="stylesheet" href="../stylesheets/stylesheet.css" media="screen"/>
  <link rel="stylesheet" href="../stylesheets/pygment_trac.css"/>
  <script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
  <script type="text/javascript" src="../javascripts/script.js"></script>
  <title>UniProt</title>
</head>

<body>

  <div class="wrapper">
    <header>
      <h1 class="title">R Notes for Bioinformatics</h1>
    </header>
    <div id="container">
      <p class="tagline">A collection of notes describing a few aspects of R for bioinformatics.</p>
      <div id="main" role="main">
        <div class="download-bar">
        <div class="inner">
          <a href="https://github.com/chernan/RNotes4Bioinformatics/tarball/gh-pages" class="download-button tar"><span>Download</span></a>
          <a href="https://github.com/chernan/RNotes4Bioinformatics/zipball/gh-pages" class="download-button zip"><span>Download</span></a>
          <a href="https://github.com/chernan/RNotes4Bioinformatics" class="code">View R Notes for Bioinformatics on GitHub</a>
        </div>
        <span class="blc"></span><span class="trc"></span>
        </div>
        <article class="markdown-body">
        
<h1>UniProt</h1>

<p>UniProt is the best protein database for manually curated annotations.</p>

<h2>Using biomaRt to retrieve data from UniProt</h2>

<h3>Installation</h3>

<pre><code class="r">source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite(&quot;biomaRt&quot;)
library(&quot;biomaRt&quot;)
</code></pre>

<h3>Generalities</h3>

<p>One can list available databases for biomaRt.</p>

<pre><code class="r">uniProt &lt;- useMart(biomart = &quot;unimart&quot;)
listDatasets(uniProt)
</code></pre>

<pre><code>##   dataset description version
## 1 uniprot     uniprot
</code></pre>

<h3>Constructing a request</h3>

<p>Selecting the UniProt dataset, updating the Mart that was just created.</p>

<pre><code class="r">uniProt &lt;- useDataset(&quot;uniprot&quot;, mart = uniProt)
</code></pre>

<p>Alternatively, if the dataset is known already, it can be specified during the Mart creation.</p>

<pre><code class="r">uniProt &lt;- useMart(biomart = &quot;unimart&quot;, dataset = &quot;uniprot&quot;)
</code></pre>

<p>Available filters and attributes can be accessed through corresponding methods.</p>

<pre><code class="r">listFilters(uniProt)
</code></pre>

<pre><code>##                name       description
## 1  superregnum_name  Superregnum name
## 2     proteome_name Complete proteome
## 3         accession         Accession
## 4      protein_name           Protein
## 5    length_greater          Length &gt;
## 6    length_smaller          Length &lt;
## 7  protein_evidence Protein existence
## 8           embl_id          EMBL IDs
## 9   arrayexpress_id  ArrayExpress IDs
## 10       ensembl_id       Ensembl IDs
## 11        pdbsum_id        PDBSum IDs
## 12        intact_id        IntAct IDs
## 13      interpro_id      InterPro IDs
## 14            go_id Gene Ontology IDs
## 15        gene_name         Gene name
## 16       entry_type        Entry type
## 17        organelle         organelle
## 18        plasmid_f           Plasmid
</code></pre>

<pre><code class="r">listAttributes(uniProt)
</code></pre>

<pre><code>##                       name       description
## 1                accession         Accession
## 2                     name        Entry name
## 3             protein_name      Protein name
## 4                gene_name         Gene name
## 5                 organism          Organism
## 6         protein_evidence Protein existence
## 7               entry_type            Status
## 8                    go_id             GO ID
## 9                  go_name           GO name
## 10  db2go_p__dm_primary_id          GO ID(p)
## 11 db2go_p__dm_description           GO name
## 12 db2go_f__dm_description       GO name (F)
## 13  db2go_f__dm_primary_id         GO ID (F)
## 14  db2go_c__dm_primary_id         GO ID (C)
## 15 db2go_c__dm_description       GO name (C)
## 16                 embl_id          EMBL IDs
## 17              ensembl_id       Ensembl IDs
## 18             interpro_id      InterPro IDs
## 19               pdbsum_id        PDBSum IDs
## 20                  pdb_id           PDB IDs
## 21            arrayexpress  ArrayExpress IDs
## 22                pride_id         PRIDE IDs
## 23             interact_id        IntAct IDs
## 24                comments          Comments
## 25               ec_number         Ec number
## 26                 keyword           Keyword
## 27            plasmid_name      Plasmid name
## 28          organelle_name    organelle name
</code></pre>

<p>Filters are the building blocks of requests.<br/>
Attributes are what can be returned by the requests.  </p>

<h3>Examples</h3>

<p>From a UniProt AC, request the Protein name (ID) and Gene Ontology IDs and names.</p>

<pre><code class="r">uniProt &lt;- useMart(biomart = &quot;unimart&quot;, dataset = &quot;uniprot&quot;)
getBM(attributes = c(&quot;name&quot;, &quot;go_id&quot;, &quot;go_name&quot;), values = &quot;E9HCD7&quot;, filter = &quot;accession&quot;, 
    mart = uniProt)
</code></pre>

<pre><code>##         name      go_id                    go_name
## 1 NNRE_DAPPU GO:0052856 F:NADHX epimerase activity
## 2 NNRE_DAPPU GO:0046872        F:metal ion binding
## 3 NNRE_DAPPU GO:0000166       F:nucleotide binding
</code></pre>

<p><a rel="license" href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US"><img alt="Creative Commons License" style="border-width:0" src="http://i.creativecommons.org/l/by-sa/3.0/80x15.png" /></a><br />This work by <span xmlns:cc="http://creativecommons.org/ns#" property="cc:attributionName">Celine Hernandez</span> is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-sa/3.0/deed.en_US">Creative Commons Attribution-ShareAlike 3.0 Unported License</a>.</p>

        </article>
      </div>
    </div>
    <footer>
      <div class="owner">
      <p><a href="https://github.com/chernan" class="avatar"><img src="https://0.gravatar.com/avatar/87ffdd8b5458b07e667a4e23a636b8dd?d=https%3A%2F%2Fidenticons.github.com%2F52da2d5c8b30ddc47bfbd75cd46d8a65.png&amp;s=30" width="48" height="48"/></a> <a href="https://github.com/chernan">chernan</a> maintains <a href="https://github.com/chernan/RNotes4Bioinformatics">R Notes for Bioinformatics</a></p>


      </div>
      <div class="creds">
        <small>This page generated using <a href="https://pages.github.com/">GitHub Pages</a><br/>theme by <a href="https://twitter.com/jonrohan/">Jon Rohan</a></small>
      </div>
    </footer>
  </div>
  <div class="current-section">
    <a href="#top">Scroll to top</a>
    <a href="https://github.com/chernan/RNotes4Bioinformatics/tarball/master" class="tar">tar</a><a href="https://github.com/chernan/RNotes4Bioinformatics/zipball/master" class="zip">zip</a><a href="" class="code">source code</a>
    <p class="name"></p>
  </div>

  
</body>


</html>
